check_test.go

```diff
@@ -186,6 +186,57 @@ var testcases = map[string]struct {
 			})
 		},
 	},
+	"CreateRepositoryGrep": {
+		fixture: "check/create_repo_grep",
+		body: func(c *ol.Client) (*ol.Check, error) {
+			return c.CreateCheckRepositoryGrep(ol.CheckRepositoryGrepCreateInput{
+				CheckCreateInput: checkCreateInput,
+				DirectorySearch:  true,
+				Filepaths:        []string{"**/hello.go"},
+				FileContentsPredicate: &ol.PredicateInput{
+					Type:  ol.PredicateTypeEnumExists,
+					Value: "",
+				},
+			})
+		},
+	},
+	"UpdateRepositoryGrep": {
+		fixture: "check/update_repo_grep",
+		body: func(c *ol.Client) (*ol.Check, error) {
+			return c.UpdateCheckRepositoryGrep(ol.CheckRepositoryGrepUpdateInput{
+				CheckUpdateInput: checkUpdateInput,
+				DirectorySearch:  ol.Bool(true),
+				Filepaths:        []string{"**/go.mod"},
+				FileContentsPredicate: &ol.PredicateInput{
+					Type:  ol.PredicateTypeEnumExists,
+					Value: "",
+				},
+			})
+		},
+	},
+	"UpdateRepositoryGrepMissingDirectorySearch": {
+		fixture: "check/update_repo_grep_missing_directory_search",
+		body: func(c *ol.Client) (*ol.Check, error) {
+			return c.UpdateCheckRepositoryGrep(ol.CheckRepositoryGrepUpdateInput{
+				CheckUpdateInput: checkUpdateInput,
+				Filepaths:        []string{"**/go.mod"},
+				FileContentsPredicate: &ol.PredicateInput{
+					Type:  ol.PredicateTypeEnumExists,
+					Value: "",
+				},
+			})
+		},
+	},
+	"UpdateRepositoryGrepMissingFileContentsPredicate": {
+		fixture: "check/update_repo_grep_missing_file_contents_predicate",
+		body: func(c *ol.Client) (*ol.Check, error) {
+			return c.UpdateCheckRepositoryGrep(ol.CheckRepositoryGrepUpdateInput{
+				CheckUpdateInput: checkUpdateInput,
+				DirectorySearch:  ol.Bool(true),
+				Filepaths:        []string{"**/go.mod"},
+			})
```

Comment:
I think what we actually need to test here is being able to "unset" `FileContentsPredicate` when its already set.  This is the usecase Terraform will do when its removed from the config and it needs to be turn from an existing predicate to "null" - basically "unset" a predicate.  I'm not confident our code as written can do that.

If this doesn't make sense lets sync on this Wednesday

Reply:
Yeah, you're right, I see that FileContentsPredicate is unchanged when I run the mutation.  I'll take a look at it today to see if I can make the necessary changes.  I followed the pattern of other check types so sounds like we need to make this change in a lot of places.